packages/database/src/migrations/20210921051858-LESMISRepititionRatesDistrict-modifies-data.js

```diff
@@ -0,0 +1,276 @@
+      {
+        transform: 'updateColumns',
+        insert: {
+          organisationUnitCode: '=$organisationUnit',
+          value: '=$value',
+        },
+        exclude: ['organisationUnit', 'dataElement', 'period'],
+      },
+    ],
+  },
+});
+
+const getGpiReport = (reportCode, dataElement) => ({
+  id: generateId(),
+  code: reportCode,
+  config: {
+    fetch: {
+      dataElements: dataElement,
+      aggregations: [
+        {
+          type: 'FINAL_EACH_YEAR',
+          config: {
+            dataSourceEntityType: 'sub_district',
+            aggregationEntityType: 'requested',
+          },
+        },
+      ],
+    },
+    transform: [
+      {
+        transform: 'updateColumns',
+        insert: {
+          organisationUnitCode: '=$organisationUnit',
+          '=$dataElement': '=divide($value, 100)',
+        },
+        exclude: ['dataElement', 'value', 'organisationUnit'],
+      },
+      {
+        transform: 'mergeRows',
+        groupBy: ['organisationUnitCode', 'period'],
+        using: 'single',
+      },
+      {
+        transform: 'updateColumns',
+        insert: {
+          value: `=divide($${dataElement[0]},$${dataElement[1]})`,
+        },
+        exclude: [dataElement[0], dataElement[1]],
+      },
+    ],
+  },
+});
+
+const getMapOverlay = (name, reportCode) => ({
+  id: reportCode,
+  name,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode,
+  },
+  presentationOptions: {
+    scaleType: 'performanceDesc',
+    displayType: 'shaded-spectrum',
+    measureLevel: 'SubDistrict',
+    valueType: 'percentage',
+    scaleBounds: {
+      left: {
+        max: 'auto',
+      },
+      right: {
+        min: 'auto',
+      },
+    },
+    periodGranularity: 'one_year_at_a_time',
+  },
+  countryCodes: '{"LA"}',
+  projectCodes: '{laos_schools}',
+});
+
+const getGpiMapOverlay = (name, reportCode) => ({
+  id: reportCode,
+  name,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode,
+  },
+  presentationOptions: {
+    scaleType: 'gpi',
+    displayType: 'shaded-spectrum',
+    measureLevel: 'SubDistrict',
+    scaleBounds: {
+      left: {
+        max: 0,
+      },
+      right: {
+        min: 2,
+      },
+    },
+    periodGranularity: 'one_year_at_a_time',
+  },
+  countryCodes: '{"LA"}',
+  projectCodes: '{laos_schools}',
+});
+
+const getPermissionGroupId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM permission_group WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const getSpecificReport = (reportCode, dataElement, GPI) => {
+  if (GPI) {
+    return getGpiReport(reportCode, dataElement);
+  }
+  return getReport(reportCode, dataElement);
+};
```

Comment:
It's not necessary to change this but just letting you know this could be made more concise as follows.
```suggestion
const getSpecificReport = (reportCode, dataElement, GPI) => GPI ? getGpiReport(reportCode, dataElement): getReport(reportCode, dataElement);
```

Reply:
Thanks Tom, good suggestion.